### ISSUE TITLE
JSON Schema Validation errors for Params prevent DAG from importing and errors are not shown

### ISSUE THREAD

**Issue by Blizzke** [apache/airflow#29828]
### Apache Airflow version

2.5.1

### What happened

When declaring a DAG Param, any resulting `jsonschema.exceptions.SchemaErrors` are not shown in the GUI as a DAG import error, nor visible via `airflow dags list-import-errors` because the exception is not caught in the `models/dagbag.py` ` _process_modules`-function (only `ParamValidationError` is)

### What you think should happen instead

The function should not only capture `ParamValidationError`s, but also `jsonschema.exceptions.SchemaError`. and let you know why the DAG is not being imported.

### How to reproduce

Add a `Param` to your dag, but use a wrong configuration. In  our case the developer mistakingly used `schema` instead of `type`. 

```
"release": Param(
  default=None,
  description="Release to run the update for",
  schema="string"
 )
```

The `schema` line should've been `type=["string", "null"]`.

### Operating System

Debian

### Deployment

Docker-Compose

### Deployment details

Original airflow docker container (2.5.1 - python 3.10)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Blizzke** [comment]
Screenshot of the output during `airflow db upgrade` (only place where it was shown that I could find during debugging this):
![params](https://user-images.githubusercontent.com/231572/222085182-b158c685-0f0e-4291-b334-01236b3554ae.png)

**Comment by SamWheating** [comment]
This was inadvertently fixed in https://github.com/apache/airflow/pull/29056, where the `Except` in `_process_modules` was modified to catch any exceptions. 

Can confirm that these issues are correctly surfaced when running Airflow from `main`:
<img width="1881" alt="image" src="https://user-images.githubusercontent.com/16950874/222573097-6dffd0e7-1acf-4454-95da-0eb305c88abe.png">

**Comment by potiuk** [comment]
Fixed in #29056 then. Closing as essentially duplicate of it.
